### ISSUE TITLE
Throwing Exception When Calling DotEnv.configure

### ISSUE THREAD

**Issue by azamsharpschool** [thebarndog/swift-dotenv#25]
Dotenv.configure() is throwing exception 

` Fatal error: 'try!' expression unexpectedly raised an error: SwiftDotenv.Dotenv.LoadingFailure.environmentFileIsMissing`

even though I have a file called .env with keys and values.

**Comment by thebarndog** [comment]
@azamsharpschool what's the location of your `.env` file? The default path is `.env` but your file might not be located there. Assuming that you're developing a script? If you're developing for an iOS application or another platform, then you'll have change the default file path from `.env` to wherever the file lives but then you'll have to include the file in your app bundle which defeats the purpose of using this library in the first place.

I think it's not clear from the README and I can update it because this isn't the first time it's come up but this library is for _scripting only_ when using in conjunction with a `.env` file. If you just want to read environment values from the underlying process, you can (it just proxies to `processInfo.environment`) but loading a `.env` file into an application would mean you have to include it in your bundle and in your version control.

**Comment by minsoe** [comment]
So, where should we put this `.env` file iOS? the same location as Xcode project file?

**Comment by thebarndog** [comment]
No, you don't use .env files for iOS applications; it's for scripting only. Think about it, it doesn't make sense. In order for the iOS application to be able to see the environment file, it would have to be included in the bundle.  And even if you manage to keep it out of version control, you're still loading that file into the application bundle that gets distributed to users.

**Comment by doozMen** [comment]
I have the same issue and after some debugging found out that it works with swift version > 6.0 but not lower. I think it calls a dynamic library or something that makes the tool depend on the swift version installed. The colleague of mine having the issue is on Sonoma using xcode 15.3 and I'm on Sequoia using xcode 16. That is the only difference and this error messages pops up.

**Comment by thebarndog** [comment]
Hmm @doozMen that's interesting, and difficult to replicate. I'll try to take a look soon and see if I can figure out why that's the case

**Comment by doozMen** [comment]
It originates in the parser. I wrote my own regex and that works just fine. Will send a pr with what I got now to maybe tackle this issue upstream if you agree.

**Comment by thebarndog** [comment]
Sure by all means!

**Comment by TheFern2** [comment]
> No, you don't use .env files for iOS applications; it's for scripting only. Think about it, it doesn't make sense. In order for the iOS application to be able to see the environment file, it would have to be included in the bundle. And even if you manage to keep it out of version control, you're still loading that file into the application bundle that gets distributed to users.

I have an iOS app that I am working on, where it uses custom oauth2 flow, and I don't want to run a backend server to get tokens and refresh them while I develop. So I am using my PAT for development, so yes, there are use cases where you use for development, obviously not for production. I ask again where should the .env file be located?

**Comment by thebarndog** [comment]
@doozMen any progress on this? I tried replicating it myself using swift version 5.10 but unfortunately was unable to get my tests to fail. I'm considering closing this issue out since 6.0 has been out for some time now but happy to keep it open if you've uncovered anything!

**Comment by thebarndog** [comment]
Closing due to inactivity
